Enterprise-CMCS/macpro-onemac
# OY2 25156: Disable Withdraw Formal RAI Response Action - CMS

Story: https://qmacbis.atlassian.net/browse/OY2-25156
Endpoint: https://d3nsbarmz97n53.cloudfront.net/

### Details
After trying to automagically disable the RAI response withdraw using the SEA Tool status.... it was realized that sometimes, you gotta just let someone correct their mistakes in the interface. 

Also, "RAI Response Withdraw Enabled" is too fragile of a status, because any update to SEA Tool will change the status and remove the enabling.  Was decided that the "RAI Response Withdraw Enabled" status OVERRIDES other status changes until it is either disabled by the CMS user or the RAI Response Withdraw Request is made by the state user.

### Changes
- When a package is in "RAI Response Withdraw Enabled" status, the CMS user is shown a "disable" action on the details page (CMS users do not have dashboard actions)
- If there is an RAI Response in "RAI Response Withdraw Enabled" status, the package status is forced to be "RAI Response Withdraw Enabled."
- Disable action takes user to a similar form to the enable action request.
- Disable action saves the details in the admin change section of the RAI Response event
- Disable action RETURNS THE PACKAGE TO THE PREVIOUS STATUS (yes.... this was the hard part!)
- found and fixed a bug with the package builder -- there was a mismatch between the event configs and the package configs and the events were using the parent package's configurations.  This caused side effects like the "Initial Submission" date getting set to the latest RAI Response submission date.
- normalized event naming to remove duplication in the processing of event "families" (like initial submission, rai responses)
- found and fixed bug regarding reverse chrono attachments (added reverseChrono to the list of attributes being given signed URLs)

### Implementation Notes
- had to refactor the package builder

### Test Plan
1. Login as a cms user
2. Locate a package in "RAI Response Withdraw Enabled" status
3. Use the AWS console to fake a SEA Tool event and verify it did not update the package status
4. Go to the package details
5. Verify the "Disable RAI Response Withdraw" action is available
6. Disable the RAI Response Withdraw
7. Verify the RAI Response is no longer enabled for withdrawal and the most recent status is shown (yes, might mean a new status if there was a change from SEA Tool)
